default/AI/AIstate.py

```diff
@@ -120,8 +127,28 @@ def __setstate__(self, state_dict):
         for odict_attrib in ['colonisablePlanetIDs', 'colonisableOutpostIDs']:
             if dict_attrib not in state_dict:
                 self.__dict__[odict_attrib] = odict()
+        if 'uuid' not in state_dict:
+            self.uuid = uuid4()
+        if 'turn_hashes' not in state_dict:
```

Comment:
Mismatch in attributes names is issue, will fix it tomorrow.
Other thing I will do is change game uuid4() to timestamp or time string. It will be unique enough and can be used for sorting.